addons/website/models/website.py

```diff
@@ -527,9 +527,9 @@ def viewref(self, view_id, raise_if_not_found=True):
             if views:
                 view = views.filter_duplicate()
             else:
-                view = self.env.ref(view_id)
+                view = self.env.ref(view_id, raise_if_not_found=raise_if_not_found)
                 # self.env.ref might return something else than an ir.ui.view (eg: a theme.ir.ui.view)
-                if view._name != 'ir.ui.view':
+                if not view or view._name != 'ir.ui.view':
```

Comment:
if view and view._name...

Reply:
`not view or` ensure `view=None` when it's falsy
`view and` the view will keep any falsy value it had

It doesn't actually matter in this case since we only use that value in a condition after.

Reply:
Si tu n'as pas a manger devant toi ou que ce sont des légumes, alors vide ton assiette (ne fait rien)...
(qui est deja Falsy et retournait donc None)

Ca a quand meme plus de sens de dire 
"Si tu as a manger, et que ce que tu manges n'est pas des légumes, vide l'assiette "

mais mtnt que on ensure que si pas de vue on retourne un recordset vide, c ok comme ca ;)